sample/android-app/src/main/java/com/icerockdev/MainActivity.kt

```diff
@@ -1,13 +1,82 @@
 package com.icerockdev
 
 import android.os.Bundle
+import android.view.View
+import android.widget.LinearLayout
+import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.snackbar.Snackbar
+import com.icerockdev.library.PermissionsHandler
+import com.icerockdev.library.mainCoroutineContext
+import dev.icerock.moko.permissions.DeniedAlwaysException
+import dev.icerock.moko.permissions.DeniedException
+import dev.icerock.moko.permissions.Permission
+import dev.icerock.moko.permissions.PermissionsController
+import android.provider.Settings.ACTION_APPLICATION_DETAILS_SETTINGS
+import android.content.Intent
+import android.net.Uri
 
 class MainActivity : AppCompatActivity() {
 
+    private lateinit var permissionsHandler: PermissionsHandler
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-
         setContentView(R.layout.activity_main)
+
+        // Prepares the permissions controller and binds it to the activity lifecycle.
+        val permissionsController = PermissionsController().apply {
+            bind(this@MainActivity.lifecycle, supportFragmentManager)
+        }
+
+        permissionsHandler = PermissionsHandler(mainCoroutineContext, permissionsController)
+    }
+
+    fun onRequestButtonClick(view: View?) {
+        // Starts permission providing process.
+        permissionsHandler.providePermission(Permission.CAMERA, object :
+            PermissionsHandler.PermissionsProviderListener {
```

Comment:
here listener should be apply to `MainActivity` and anonymous class will be unnecessary

Reply:
Will be removed after adding the moko-mvvm. `EventsDispatcher` will be used instead of the callback.